### PR TITLE
update DSM with rbus provider

### PR DESCRIPTION
**Pull request by syed-ahmad-consult-red**

### CODE REVIEW

File: develop/integration/rdk-b-22q3/meta-rdk/recipes-containers/librbusprovider/librbusprovider.bb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+SRC_URI = "git://git@gitlab.consult.red/dave.chapman/rdk-smart-router.git;protocol=ssh;branch=main \
```
Comment (mhughesacn) [inline develop/integration/rdk-b-22q3/meta-rdk/recipes-containers/librbusprovider/librbusprovider.bb:1]: Is this a publicly accessible site?
How is it licensed?  (LICENSE set to append "" is unusual.)
(If not publicly accesible, then we have a recipe in an opensourced repo referring to somewhere private.)

File: develop/integration/rdk-b-22q3/meta-rdk/recipes-containers/dsm/dsm.bb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SUMMARY = "Downloadable Software Modules"
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=50e2d278b66b3b7b20bc165c146a2b58"
+
+SRC_URI = "git://git@gitlab.consult.red/dave.chapman/rdk-smart-router.git;protocol=ssh;branch=main"
```
Comment (mhughesacn) [inline develop/integration/rdk-b-22q3/meta-rdk/recipes-containers/dsm/dsm.bb:22]: Same comment as librbusprovider.bb
i.e. To confirm that SRC_URI is public, and presumably apache-licensed.

File: src/rbus/dsm_rbus_provider.hpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
+#ifndef DSM_RBUS_PROVIDER_H
```
Comment (mhughesacn) [inline src/rbus/dsm_rbus_provider.hpp:1]: Missing header on new file

File: src/rbus/dsm_rbus_provider.cpp [hunk 4]
```diff
@@ -0,0 +1,1040 @@
+#include "dsm_rbus_provider.hpp"
```
Comment (mhughesacn) [inline src/rbus/dsm_rbus_provider.cpp:1]: Missing header on new file

**Review by mhughesacn — COMMENTED** [review]
This repository is still laid out in a manner not usual in RDK i.e. metadata mixed with code.  The new recipes add to this.  There was a request that this would be regularised post opensourcing (BD-1138, internal), not compounded.  e.g. In my (limited!) knowledge, patching recipes is not the norm (bbappends would be).  Unfortunately, our yocto expert is on vacation or I would ask him to comment.  I will ask Sundar Subramanian (RDK) to take a look as he was involved with opensourcing, and may have some guidance before I approve.  I have some other questions as well below.

Thank you
Martin Hughes, RDK CMF Compliance Team

File: develop/integration/rdk-b-22q3/meta-rdk/recipes-containers/librbusprovider/librbusprovider.bb (hunk 1 above)
Comment (dphc1971) [reply develop/integration/rdk-b-22q3/meta-rdk/recipes-containers/librbusprovider/librbusprovider.bb:1]: This should not have been included in the pull request. We now have all recipes as pending merges into existing meta-layers. The recipes were removed by a previous pull request which I merged. We're pushing some improvements from our local copy of DSM to the hosted on here. Unfortunately I wasn't clear in stating it should only be the code under the src directory that should be pushed. Will either close this pull request, or fix so its only code updates, and there are no more recipes here ... as it should be.

File: develop/integration/rdk-b-22q3/meta-rdk/recipes-containers/dsm/dsm.bb (hunk 2 above)
Comment (dphc1971) [reply develop/integration/rdk-b-22q3/meta-rdk/recipes-containers/dsm/dsm.bb:22]: This should not have been included in the pull request. We now have all recipes as pending merges into existing meta-layers. The recipes were removed by a previous pull request which I merged. We're pushing some improvements from our local copy of DSM to the hosted on here. Unfortunately I wasn't clear in stating it should only be the code under the src directory that should be pushed. Will either close this pull request, or fix so its only code updates, and there are no more recipes here ... as it should be.

File: src/rbus/dsm_rbus_provider.hpp (hunk 3 above)
Comment (dphc1971) [reply src/rbus/dsm_rbus_provider.hpp:1]: We have an RBUS integration with DSM which will eventually be made public, but not in this pull request which is about internal fixes for DSM. We'll review the RBUS integration code when we release it.

File: src/rbus/dsm_rbus_provider.cpp (hunk 4 above)
Comment (dphc1971) [reply src/rbus/dsm_rbus_provider.cpp:1]: As Above

**Comment by syed-ahmad-consult-red** [comment]
Removing this PR.

**Comment by mhughesacn** [comment]
Thanks Dave.
Syed, I think that you just close this pull request, rather than delete it.  (Not sure if pull reqs can be deleted easily.)  You should have permission.

**Pull request closed without merging**